cmd/uplink/cmd/cp.go

```diff
@@ -144,27 +142,48 @@ func download(ctx context.Context, src fpath.FPath, dst fpath.FPath, showProgres
 		return fmt.Errorf("destination must be local path: %s", dst)
 	}
 
-	metainfo, streams, err := cfg.Metainfo(ctx)
+	project, err := cfg.GetProject(ctx)
 	if err != nil {
 		return err
 	}
+	defer func() {
+		if err := project.Close(); err != nil {
+			fmt.Printf("error closing project: %+v\n", err)
+		}
+	}()
+
+	var access libuplink.EncryptionAccess
+	copy(access.Key[:], []byte(cfg.Enc.Key))
 
-	readOnlyStream, err := metainfo.GetObjectStream(ctx, src.Bucket(), src.Path())
+	bucket, err := project.OpenBucket(ctx, src.Bucket(), &access)
```

Comment:
since you have to open the project and bucket a few times, could there be a helper that does both together for you?